src/parser/shared/metrics/apl/conditions/buffSoonPresent.tsx

```diff
@@ -0,0 +1,81 @@
+import type Spell from 'common/SPELLS/Spell';
+import { SpellLink } from 'interface';
+import { EventType } from 'parser/core/Events';
+import { Condition, tenseAlt } from '../index';
+import { formatTimestampRange, Range } from './util';
+import { TrackedBuffEvent } from 'parser/core/Entity';
+
+type BuffInformation = {
+  buffs: TrackedBuffEvent[];
+  nextBuff: number | undefined;
+  currentIndex: number;
+};
+
+type SourceOptions = 'player' | 'external' | 'any';
+
+export function buffSoonPresent(
```

Comment:
this is a bit of a scary APL condition---not because it is inherently harmful, but because it lets people write APL rules that are not actually reflective of in-game play. i'd recommend a docblock for this explaining that:

```suggestion
/**
 * Check if a buff will be present soon after the present time.
 *
 * This should ONLY be used for buffs that the player has complete control over, such as buffs tied to cooldown use.
 * DO NOT use this to "look ahead" for random procs to implement impossibly optimal APLs.
 */
export function buffSoonPresent(
```

Reply:
Added the docblock - didn't even consider that usecase but makes sense to discourage it.